src/main/java/org/locationtech/spatial4j/shape/jts/JtsGeometry.java

```diff
@@ -573,17 +573,16 @@ private static Geometry cutUnwrappedGeomInto360(Geometry geom) {
       return geom;
     assert geom.isValid() : "geom";
 
-    //TODO opt: support geom's that start at negative pages --
-    // ... will avoid need to previously shift in unwrapDateline(geom).
     List<Geometry> geomList = new ArrayList<Geometry>();
     //page 0 is the standard -180 to 180 range
-    for (int page = 0; true; page++) {
+    int startPage = (int) Math.floor((geomEnv.getMinX() + 180) / 360);
+    for (int page = startPage; true; page++) {
       double minX = -180 + page * 360;
       if (geomEnv.getMaxX() <= minX)
         break;
       Geometry rect = geom.getFactory().toGeometry(new Envelope(minX, minX + 360, -90, 90));
       assert rect.isValid() : "rect";
-      Geometry pageGeom = rect.intersection(geom);//JTS is doing some hard work
+      Geometry pageGeom = rect.intersection(geom).copy();//JTS is doing some hard work
```

Comment:
Shifting the intersection also alters the original geometry. Using the example from `JtsSpatialContextTest`, here is an overview of the value of `geom` in the `cutUnwrappedGeomInto360` loop:
- Page 0
  - `geom`: POLYGON ((181 -90, 179 -90, 179 90, 181 90, 181 -90))
  - `pageGeom`: POLYGON ((179 90, 180 90, 180 -90, 179 -90, 179 90))
  - Intersection isn't shifted because it is already in the -180 to 180 area.
- Page 1
  - `geom`: POLYGON ((181 -90, 179 -90, 179 90, 181 90, 181 -90))
  - `pageGeom`: POLYGON ((180 -90, 180 90, 181 90, 181 -90, 180 -90))
  - **Now the intersection is shifted -360 degrees**
- Page 2
  - `geom`: POLYGON ((181 -90, 179 -90, 179 90, 181 90, **-179** -90))
  - No intersection is calculated because the geometry doesn't span across this page. But notice that the original geometry has changed by shifting the intersection in the previous iteration.

But when we start on a negative page, we already shift in the first iteration and then the geometry is messed up for the following iterations. This hasn't been noticed in the tests because all tested geometries only have 2 pages (0 and 1). If it is possible to have more than 2 pages, you would see the same problem there.

Reply:
> Shifting the intersection also alters the original geometry. 

Woah; I thought the intersection produced an entirely new geometry!

I'm glad you are looking into this.  If it's not asking too much, perhaps a new test might perturb these problems.